devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -1495,6 +1514,40 @@ private void setJvmArgs(DevModeCommandLineBuilder builder) throws Exception {
 
     }
 
+    private void copySurefireVariables() {
+        if (!copySurefireVariables) {
+            return;
+        }
+
+        Plugin surefireMavenPlugin = getConfiguredPluginOrNull(ORG_APACHE_MAVEN_PLUGINS, MAVEN_SUREFIRE_PLUGIN);
+        if (surefireMavenPlugin == null) {
+            return;
+        }
+
+        Xpp3Dom config = (Xpp3Dom) surefireMavenPlugin.getConfiguration();
+        if (config != null) {
+            // we copy the maps because they can be unmodifiable
+            environmentVariables = new HashMap<>(environmentVariables);
+            copyConfiguration(config.getChild("environmentVariables"), environmentVariables);
+            systemProperties = new HashMap<>(systemProperties);
+            copyConfiguration(config.getChild("systemPropertyVariables"), systemProperties);
+        }
+    }
+
+    private void copyConfiguration(Xpp3Dom config, Map<String, String> targetMap) {
+        if (config == null) {
+            return;
+        }
+
+        for (Xpp3Dom child : config.getChildren()) {
+            String key = child.getName();
+            String value = child.getValue();
+            if (!targetMap.containsKey(key)) {
+                targetMap.put(key, value);
+            }
```

Comment:
```suggestion
            targetMap.putIfAbsent(child.getName(), child.getValue());
```
The changes are good to merge, that's a really minor comment.

Reply:
Ah, good point! I'll fix, this is really embarrassing :-D

Reply:
Done.